src/App.tsx

```diff
@@ -153,8 +164,12 @@ function AppRoot(props: { environment?: Environment }): React.ReactElement {
             value={{
               ISFavourites,
               setISFavourites,
-              reportViewSelections,
-              setReportViewSelections,
+              localStorageGeneralViews,
```

Comment:
@wooneh FYI this structural duplication is deliberately not DRY. @0qy & I know there's a better way :) , but the design needs some careful thought and we need to get this into play now. The handling of localStorage vs url--shared-state is quite tricky.